rplidar.go

```diff
@@ -149,21 +168,34 @@ func NewDevice(devicePath string) (*Device, error) {
 	defer gen.DeleteRplidar_response_device_health_t(healthInfo)
 
 	if result := driver.GetHealth(healthInfo, defaultTimeout); Result(result) != ResultOk {
+		gen.RPlidarDriverDisposeDriver(driver)
+		driver = nil
 		return nil, fmt.Errorf("failed to get health: %w", Result(result).Failed())
 	}
 
 	if int(healthInfo.GetStatus()) == gen.RPLIDAR_STATUS_ERROR {
+		gen.RPlidarDriverDisposeDriver(driver)
+		driver = nil
 		return nil, errors.New("bad health")
 	}
 
-	return &Device{
+	d := &Device{
 		driver:           driver,
 		nodeSize:         8192,
 		model:            devInfo.GetModel(),
 		serialNumber:     serialNumStr,
 		firmwareVersion:  firmwareVer,
 		hardwareRevision: hardwareRev,
-	}, nil
+	}
+
+	err := d.Start()
+	if err != nil {
+		driver = nil
+		gen.RPlidarDriverDisposeDriver(driver)
```

Comment:
I think this is backwards. You need to dispose first, before setting it to nil (as is done in the other similar cases above.)

Reply:
ugh sorry this is a mess! Shouldn't even be here!

Reply:
Taking that back - it belongs here - but you're right, the order has to be reversed.

Done.